packages/client/src/containers/MovieListPage/ReducerForFetchMoviesAndCategories.jsx

```diff
@@ -0,0 +1,101 @@
+import { apiURL } from '../../apiURL';
+
+export const initialState = {
+  movieLoading: false,
+  categoryLoading: false,
+  moviesData: {},
+  categories: [],
+  movieFetchError: '',
+  categoryFetchError: '',
+  searchText: '',
+  categoryId: null,
+  sortBy: 'rating',
+};
+
+export const handleMovieFetch = async (state, dispatchFunc) => {
+  let URI = `${apiURL()}/movies?`;
+
+  if (state.searchText) {
+    URI = `${URI}title=${state.searchText}&`;
+  }
+  if (state.sortBy) {
+    URI = `${URI}sortBy=${state.sortBy}&`;
+  }
+  if (state.categoryId) {
+    URI = `${URI}categoryId=${state.categoryId}&`;
+  }
+
+  try {
+    dispatchFunc({ type: 'fetchMovies' });
+    const response = await fetch(URI);
+    if (!response.ok) throw Error('Did not receive expected data!');
+    const result = await response.json();
+    dispatchFunc({ type: 'successMovies', payload: result });
+  } catch (err) {
+    dispatchFunc({ type: 'movieFetchError', payload: err.message });
+  }
+};
+
+export const handleCategoriesFetch = async (state, dispatchFunc) => {
+  try {
+    dispatchFunc({ type: 'fetchCategories' });
+    const response = await fetch(`${apiURL()}/categories`);
+    if (!response.ok) throw Error('Did not receive expected data!');
+    const result = await response.json();
+    dispatchFunc({ type: 'successCategories', payload: result });
+  } catch (err) {
+    dispatchFunc({ type: 'categoryFetchError', payload: err.message });
+  }
+};
+
+export const ReducerForFetchMoviesAndCategories = (state, action) => {
+  switch (action.type) {
```

Comment:
As we discussed, a reducer as such is not a bad idea at all. 
However, as we agreed, if we need context and reducer for both movies and categories, then they should be separated. However, I do not see why categories context would be necessary, but you can try to prove that it is! :)